internal/api/option/auth.go

```diff
@@ -435,9 +435,9 @@ func (a *authInterceptor) verifyIDPIssuedToken(
 	c := claims{}
 	if a.oidcTokenVerifyFn == nil {
 		verifierMu.Lock()
-		defer verifierMu.Unlock()
 		if a.oidcTokenVerifyFn == nil {
 			a.oidcTokenVerifyFn = newMultiClientVerifier(ctx, a.cfg)
+			verifierMu.Unlock()
```

Comment:
Think this should be something like:

```go
verifierMu.Lock()
if a.oidcTokenVerifyFn == nil {
    a.oidcTokenVerifyFn = newMultiClientVerifier(ctx, a.cfg)
}
verifier := a.oidcTokenVerifyFn
verifierMu.Unlock()

if verifier == nil {
    return c, errors.New(
        "could not validate token, possibly due to a transient network " +
            "error; if the problem persists, check your OpenID Connect " +
            "configuration",
    )
}
```

Reply:
607b9116de35c5c1ae1037ad1c8231254a853f4f